lib/std/json/static.zig

```diff
@@ -34,6 +34,16 @@ pub const ParseOptions = struct {
     /// The default for `parseFromTokenSource` with a `*std.json.Reader` is `std.json.default_max_value_len`.
     /// Ignored for `parseFromValue` and `parseFromValueLeaky`.
     max_value_len: ?usize = null,
+
+    /// If false, calling any parseFromSlice* or parseFromTokenSource* functions
+    /// where the given json object is only partially parsed is an assertion failure.
+    /// Setting this to true allows parseFrom* functions to parse only part of a json.
+    allow_partial: bool = false,
+
+    /// If false, dynamic elements will be duplicated when parsing,
+    /// even if they could be referenced in the buffer.
+    /// This is useful when parsing using a temporary Reader buffer.
+    trust_buffer: bool = true,
```

Comment:
How about a field `allocate: AllocWhen` and then all the places that test `options.trust_buffer`  can just pass `options.allocate` instead, i.e.
```zig
const token = try source.nextAllocMax(allocator, options.allocate, options.max_value_len.?);
```

Reply:
That is cleaner, change made.